spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/TaskConfigurationProperties.java

```diff
@@ -16,32 +16,34 @@
 
 package org.springframework.cloud.dataflow.server.service.impl;
 
-import javax.validation.constraints.NotBlank;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.cloud.dataflow.core.DataFlowPropertyKeys;
+import org.springframework.util.StringUtils;
 import org.springframework.validation.annotation.Validated;
 
-
 /**
  * Properties used to define the behavior of tasks created by Spring Cloud Data Flow.
  *
  * @author Glenn Renfro
  * @author David Turanski
+ * @author Chris Schaefer
  */
 @Validated
 @ConfigurationProperties(prefix = TaskConfigurationProperties.TASK_PREFIX)
 public class TaskConfigurationProperties {
+	private static final Logger LOGGER = LoggerFactory.getLogger(TaskConfigurationProperties.class);
 
 	public static final String TASK_PREFIX = DataFlowPropertyKeys.PREFIX + "task";
 
+	@Deprecated
 	public static final String COMPOSED_TASK_RUNNER_NAME = "composed-task-runner";
 
-	/**
-	 * The task application uri to be used for the composed task runner.
-	 */
-	@NotBlank
-	private String composedTaskRunnerUri = "composed-task-runner";
+	@Autowired
```

Comment:
In some of our code we make the assumption that composedTaskRunnerConfigurationProperties can be null.    With that assumption should we mark this Autowired as optional.